calculator/src/test/java/edu/nextstep/camp/calculator/CalculatorViewModelTest.kt

```diff
@@ -0,0 +1,66 @@
+package edu.nextstep.camp.calculator
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.google.common.truth.Truth.assertThat
+import edu.nextstep.camp.calculator.domain.Expression
+import edu.nextstep.camp.calculator.domain.Operator
+import io.mockk.every
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class CalculatorViewModelTest {
+    @get:Rule
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var viewModel: CalculatorViewModel
+
+    @Before
+    fun setUp() {
+        viewModel = CalculatorViewModel()
+    }
+
+    @Test
+    fun `숫자가 입력되면 수식에 추가되고 변경된 수식을 보여줘야 한다`() {
+        // when
+        viewModel.addToExpression(1)
+
+        // then
+        assertThat(viewModel.calculatorText.getOrAwaitValue()).isEqualTo("1")
+    }
+
+    @Test
+    fun `연산자가 입력되면 수식에 추가되고 변경된 수식을 보여줘야 한다`() {
+        // when
+        viewModel.addToExpression(1)
+        viewModel.addToExpression(Operator.Plus)
+
+        // then
+        assertThat(viewModel.calculatorText.getOrAwaitValue()).isEqualTo("1 +")
+    }
+
+    @Test
+    fun `지우기가 실행되면 수식의 마지막이 지워지고 변경된 수식을 보여줘야 한다`() {
+        // when
+        viewModel.addToExpression(1)
+        viewModel.removeLast()
+
+        // then
+        assertThat(viewModel.calculatorText.getOrAwaitValue()).isEqualTo("")
+    }
+
+    @Test
+    fun `계산이 실행되면 계산기에 의해 계산되고 결과를 화면에 보여줘야 한다`() {
+        viewModel.addToExpression(1)
+        viewModel.addToExpression(Operator.Plus)
+        viewModel.addToExpression(2)
```

Comment:
현재 테스트 가능한 ViewModel의 Expression 상태를 만들기 위해 addToExpression 이라는 함수를 통해
Expression 데이터를 구성하고 있습니다. 

이유는 Expression이 ViewModel 내부에 private val로 선언되어 있기 때문입니다.
그렇기에 Expression을 조작하여 ViewModel를 테스트 하기가 어려운 구조입니다.
또한 addToExpression 로직이 어떤 이유에서 버그가 생겨있는 상태라면 given 절이 원하는 형태의 모습으로 만들어지지 않을것 입니다.

이를 해결하기 위해서는 Expression을 조작해서 ViewModel에 넣어주는 형태로 ViewModel를 구성하면
필요한 형태의 Expression을 구성하여 유연하게 테스트를 하실 수 있습니다. 

```suggestion
        val expression = Expression(listOf("1", Operator.Plus, "2"))
        val viewModel = CalculatorViewModel(expression)
```

위 구조로 테스트를 하실수 있게 CalculatorViewModel 클래스를 변경시켜보세요! 
Testable한 클래스를 만들기 위해서 내부의 객체생성(의존성)은 없애고
외부에서 주입(Dependency Injection, DI)하는 형태로 변경시켜주시면 좋습니다. 😊